### PR TITLE
ENH: added option to train on complete dataset

### LINKED ISSUE #205: ENH: Allow model training on complete dataset
**Improvement Description**
Allowing training of estimator on complete dataset (`test_size=0.0`) while raising warning of interpretation of returned model metrics.

**Current Behavior**
Running `classify_samples` or `split_table` with `test_size=0.0` returns Plugin Error: `Parameter 'test_size' received 0.0 as an argument, which is incompatible with parameter type: Float % Range(0.0, 1.0, inclusive_start=False)`.
While option for `test_size=0.0` is specifically configured in `utilities.py` L159 - L161.

**Proposed Behavior**
Allow training of classifier with `test_size=0.0` while raising warning of the returned model performance metrics.

### PR DESCRIPTION
**Pull request by adamovanja**
closes #205 allowing model training on complete dataset (`test_size=0.0`)

### CODE REVIEW

**Review by nbokulich — CHANGES_REQUESTED** [review]
Thanks @adamovanja ! I just did a high-level review, @misialq will take over for testing

This looks good to me, I just have some minor comments/questions. The only main suggestion I have is about making the parameter a union type to explicitly "disable" instead of setting test_size=0.

File: q2_sample_classifier/assets/index.html [hunk 1]
```diff
@@ -6,6 +6,28 @@
...
+    <div class="panel-heading" role="tab" id="warnings-heading">
+      <h4 class="panel-title">
+        <a role="button" data-toggle="collapse" data-parent="#warnings" href="#warnings-list" aria-expanded="true" aria-controls="warnings-list">
+          Warnings (click here to collapse/expand):
+        </a>
+      </h4>
+    </div>
+    <div id="warnings-list" class="panel-collapse collapse in" role="tabpanel" aria-labelledby="warnings-heading">
+      <div class="alert alert-warning col-md-12">
+        <p><strong>"Displayed model performance values were calculated on the train set because estimator was
```
Comment (misialq) [inline q2_sample_classifier/assets/index.html:21]: I'd say the quotes around the message are not necessary as they also are displayed in the visualisation.

**Review by misialq — COMMENTED** [review]
All works as expected 👍

File: q2_sample_classifier/plugin_setup.py [hunk 2]
```diff
@@ -450,7 +450,8 @@
         'missing_samples': parameters['base']['missing_samples'],
         'vmin': Float | Str % Choices(['auto']),
         'vmax': Float | Str % Choices(['auto']),
-        'palette': Str % Choices(_custom_palettes().keys())},
+        'palette': Str % Choices(_custom_palettes().keys()),
+        'warn_test': Bool},
```
[the remark this answers is not in the document]
Comment (adamovanja) [reply q2_sample_classifier/plugin_setup.py:454]: warning is only made if `test_size==0`, in all other cases (`test_size>0`) there is no apparent reason to display the warning.

File: q2_sample_classifier/utilities.py [hunk 3]
```diff
@@ -838,3 +841,14 @@ def _warn_feature_selection():
...
          'different estimator model.'))
     warnings.warn(warning, UserWarning)
+
+
+def _warn_zero_test_split():
+    warning = (
+        ('Using test_size = 0.0, you are using your complete dataset for '
+         'fitting the estimator. Hence, any returned model evaluations are '
+         'based on that same training dataset and are not representative of '
+         "your model's performance on a previously unseen dataset. Please "
```
[the remark this answers is not in the document]
Comment (adamovanja) [reply q2_sample_classifier/utilities.py]: done :)

**Comment by adamovanja** [comment]
thank you both for the review. I adjusted the quotes and am happy to adjust the `test_size` change if still required.

File: q2_sample_classifier/plugin_setup.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (nbokulich) [reply q2_sample_classifier/plugin_setup.py:454]: so instead I recommend just always warning if test_size==0 and do not expose the option to disable this

File: q2_sample_classifier/plugin_setup.py [hunk 4]
```diff
@@ -102,7 +102,7 @@
         'missing_samples': Str % Choices(['error', 'ignore'])},
     'splitter': {
         'test_size': Float % Range(0.0, 1.0, inclusive_end=False,
```
[the remark this answers is not in the document]
Comment (nbokulich) [reply q2_sample_classifier/plugin_setup.py:104]: fair point, works for me

[the remark this answers is not in the document]
Comment (adamovanja) [reply q2_sample_classifier/plugin_setup.py:104]: okay, I will leave as is

**Comment by adamovanja** [comment]
I removed the warning_msg parameter from being exposed - so all the previously raised points are resolved now.

File: q2_sample_classifier/utilities.py [hunk 5]
```diff
@@ -844,11 +854,9 @@ def _warn_feature_selection():
 
 
 def _warn_zero_test_split():
-    warning = (
+    return \
```
Comment (nbokulich) [inline q2_sample_classifier/utilities.py:857]: you could drop the `\` and also the parenthesis for a cleaner look 😎

**Comment by nbokulich** [comment]
Hi @adamovanja looks good to me, I approve and just had one stylistic suggestion you can take it or leave it.

Assigning to you @thermokarst if you want to review changes addressing your comments...

File: q2_sample_classifier/utilities.py (hunk 5 above)
Comment (adamovanja) [reply q2_sample_classifier/utilities.py]: done

**Review by thermokarst — APPROVED** [review]
LGTM!

**Review by nbokulich — APPROVED** [review]

**Pull request merged**